core/src/main/java/com/linecorp/armeria/server/DefaultServiceBuilder.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    @Override
+    public ServiceBuilder requestTimeoutMillis(long requestTimeoutMillis) {
+        this.requestTimeoutMillis = validateRequestTimeoutMillis(requestTimeoutMillis);
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder maxRequestLength(long maxRequestLength) {
+        this.maxRequestLength = validateMaxRequestLength(maxRequestLength);
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder verboseResponses(boolean verboseResponses) {
+        this.verboseResponses = verboseResponses;
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder requestContentPreviewerFactory(ContentPreviewerFactory factory) {
+        requestContentPreviewerFactory = requireNonNull(factory, "factory");
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder responseContentPreviewerFactory(ContentPreviewerFactory factory) {
+        responseContentPreviewerFactory = requireNonNull(factory, "factory");
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder contentPreview(int length) {
+        return contentPreview(length, ArmeriaHttpUtil.HTTP_DEFAULT_CONTENT_CHARSET);
+    }
+
+    @Override
+    public ServiceBuilder contentPreview(int length, Charset defaultCharset) {
+        return contentPreviewerFactory(ContentPreviewerFactory.ofText(length, defaultCharset));
+    }
+
+    @Override
+    public ServiceBuilder contentPreviewerFactory(ContentPreviewerFactory factory) {
+        requestContentPreviewerFactory(factory);
+        responseContentPreviewerFactory(factory);
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder accessLogFormat(String accessLogFormat) {
+        return accessLogWriter(AccessLogWriter.custom(requireNonNull(accessLogFormat, "accessLogFormat")),
+                               true);
+    }
+
+    @Override
+    public ServiceBuilder accessLogWriter(AccessLogWriter accessLogWriter,
+                                          boolean shutdownOnStop) {
+        this.accessLogWriter = requireNonNull(accessLogWriter, "accessLogWriter");
+        shutdownAccessLogWriterOnStop = shutdownOnStop;
+        return this;
+    }
+
+    @Override
+    public <T extends Service<HttpRequest, HttpResponse>, R extends Service<HttpRequest, HttpResponse>>
+    ServiceBuilder decorator(Function<T, R> decorator) {
+        requireNonNull(decorator, "decorator");
+
+        @SuppressWarnings("unchecked")
+        final Function<Service<HttpRequest, HttpResponse>, Service<HttpRequest, HttpResponse>> castDecorator =
+                (Function<Service<HttpRequest, HttpResponse>, Service<HttpRequest, HttpResponse>>) decorator;
+
+        if (this.decorator != null) {
+            this.decorator = this.decorator.andThen(castDecorator);
+        } else {
+            this.decorator = castDecorator;
+        }
+
+        return this;
+    }
+
+    public Service<HttpRequest, HttpResponse> decorate(Service<HttpRequest, HttpResponse> service) {
```

Comment:
The non-overridden public methods in this class can be package-private because this class is package-private.